server/ServerApp.cpp

```diff
@@ -4538,9 +4538,15 @@ void ServerApp::PostCombatProcessTurns() {
     // UniverseObjects will have effects applied to them this turn, allowing
     // (for example) ships to have max fuel meters greater than 0 on the turn
     // they are created.
-    m_universe.ApplyMeterEffectsAndUpdateMeters(context, false);
+    std::vector<int> newly_created_ids;
+    for (auto& object : m_universe.Objects().all<UniverseObject>()) {
```

Comment:
probly can use `Objects().findIDs(...)` where you pass in a lambda that checks if the object should be included.